addons/website_discuss_room/static/src/js/chat_room.js

```diff
@@ -0,0 +1,127 @@
+      * - 'check-full', check if the chat room is full before joining
+      * - 'attach-to', a JQuery selector of the element on which we will add the discuss
+      *                iframe. If nothing is specified, it will open a modal instead.
+      * - 'default-username': the username to use in the chat room
+      * - 'room-server': the domain name of the Chat room server to use
+      */
+    start: async function () {
+        await this._super.apply(this, arguments);
+        this.roomName = this.$el.data('room-name');
+        this.chatRoomId = parseInt(this.$el.data('chat-room-id'));
+        // automatically open the current room
+        this.autoOpen = parseInt(this.$el.data('auto-open') || 0);
+        // before joining, perform a RPC call to verify that the chat room is not full
+        this.checkFull = parseInt(this.$el.data('check-full') || 0);
+        // query selector of the element on which we attach the room iframe
+        // if not defined, the widget will pop in a modal instead
+        this.attachTo = this.$el.data('attach-to') || false;
+        // default username for the room
+        this.defaultUsername = this.$el.data('default-username') || false;
+
+        this.roomUrl = this.$el.data('room-url');
+
+        this.maxCapacity = parseInt(this.$el.data('max-capacity')) || Infinity;
+
+        if (this.autoOpen) {
+            await this._onChatRoomClick();
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+      * Click on a chat room to join it.
+      *
+      * @private
+      */
+    _onChatRoomClick: async function () {
+        if (this.checkFull) {
+            // maybe we didn't refresh the page for a while and so we might join a room
+            // which is full, so we perform a RPC call to verify that we can really join
+            let isChatRoomFull = await rpc('/chat_room/is_full', { room_name: this.roomName });
+
+            if (isChatRoomFull) {
+                window.location.reload();
+                return;
+            }
+        }
+
+        if (this.attachTo) {
+            // attach the discuss iframe on the given parent node
+            let $parentNode = $(this.attachTo);
+            $parentNode.find("iframe").trigger("empty");
+            $parentNode.empty();
+
+            const iframe = document.createElement("iframe");
+            iframe.style = "height: 100%; width: 100%; border: 0px;"
+            iframe.src = this.roomUrl;
+            $parentNode.append(iframe);
+        } else {
+            // create a model and append the Jitsi iframe in it
+            let $jitsiModal = $(renderToElement('chat_room_modal', {}));
+            $("body").append($jitsiModal);
+            $jitsiModal.modal('show');
+
+            let jitsiRoom = await this._joinJitsiRoom($jitsiModal.find('.modal-body'));
```

Comment:
aie crash incoming ? `_joinJijtsiRoom` is not available here, not sure if we even can pass here but should probably be removed as I see this part in `webiste_jitsi`

Reply:
Wow I did a nice copy past here mybad :/
Since it is a case that is not used (don't use the attachTo option to go to a modal) You need to manually remove the t-set from meet or exhibitor if you want to test it.
Note that I have change many stuff for this part (and also in the jitsi extension)